Expose interfaces only where they should be.

In #12353, I added `Exposed=(Window,Worker)` to all interfaces, to preserve our previous behaviour. We need to go through all the interfaces, check in the specification what the correct value is, and update our implementation.
- Code: `components/script/dom/webidls/`.
- Test: `tests/wpt/mozilla/tests/mozilla/interfaces.html`, `tests/wpt/mozilla/tests/mozilla/interfaces.worker.js` and probably some others.

It would be quite difficult to review a PR that changed all interfaces at once, so please avoid doing that.

I checked

 CSSStyleDeclaration.webidl
 ElementCSSInlineStyle.webidl
 ElementContentEditable.webidl
 EventHandler.webidl
 Screen.webidl
 StyleSheetList.webidl
 StyleSheet.webidl
 Client.webidl

I had issues with the following: 
 * BrowserElement.webidl no specs
 
* ProgressEvent.webidl 
```
 WebIDL.WebIDLError: error: Unknown [Exposed] value SharedWorker, /home/tim/dev/servo/components/script/dom/webidls/ProgressEvent.webidl line 17:0
interface ProgressEvent : Event {
```

* StorageEvent.webidl and Storage.webidl
```
error[E0308]: mismatched types
    --> /home/tim/dev/servo/target/debug/build/script-2db9432a6e2c7798/out/Bindings/StorageEventBinding.rs:1031:83
     |
1031 |         let result: Result<Root<StorageEvent>, Error> = StorageEvent::Constructor(&global, arg0, &arg1);
     |                                                                                   ^^^^^^^ expected struct `dom::globalscope::GlobalScope`, found struct `dom::bindings::js::Root`
     |
     = note: expected type `&dom::globalscope::GlobalScope`
                found type `&dom::bindings::js::Root<dom::window::Window>`

error: aborting due to previous error

error: Could not compile `script`.
```

--- a/components/script/dom/webidls/CSSStyleDeclaration.webidl
+++ b/components/script/dom/webidls/CSSStyleDeclaration.webidl
@@ -8,7 +8,7 @@
  * Copyright © 2013 W3C® (MIT, ERCIM, Keio, Beihang), All Rights Reserved.
  */
 
-[Exposed=(Window, Worker)]
+[Exposed=Window]
 interface CSSStyleDeclaration {
   [SetterThrows]
            attribute DOMString cssText;

--- a/components/script/dom/webidls/Client.webidl
+++ b/components/script/dom/webidls/Client.webidl
@@ -4,7 +4,7 @@
 
 // https://w3c.github.io/ServiceWorker/#client
 
-[Pref="dom.serviceworker.enabled", Exposed=(Window,Worker)]
+[Pref="dom.serviceworker.enabled", Exposed=ServiceWorker]
 interface Client {
   readonly attribute USVString url;
   readonly attribute FrameType frameType;

--- a/components/script/dom/webidls/ElementCSSInlineStyle.webidl
+++ b/components/script/dom/webidls/ElementCSSInlineStyle.webidl
@@ -4,7 +4,7 @@
 
 //http://dev.w3.org/csswg/cssom/#elementcssinlinestyle
 
-[NoInterfaceObject, Exposed=(Window,Worker)]
+[NoInterfaceObject, Exposed=Window]
 interface ElementCSSInlineStyle {
   [SameObject/*, PutForwards=cssText*/] readonly attribute CSSStyleDeclaration style;
 };

--- a/components/script/dom/webidls/ElementContentEditable.webidl
+++ b/components/script/dom/webidls/ElementContentEditable.webidl
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // https://html.spec.whatwg.org/multipage/#elementcontenteditable
-[NoInterfaceObject, Exposed=(Window,Worker)]
+[NoInterfaceObject, Exposed=Window]
 interface ElementContentEditable {
   // attribute DOMString contentEditable;
   // readonly attribute boolean isContentEditable;

--- a/components/script/dom/webidls/EventHandler.webidl
+++ b/components/script/dom/webidls/EventHandler.webidl
@@ -25,7 +25,7 @@ callback OnBeforeUnloadEventHandlerNonNull = DOMString? (Event event);
 typedef OnBeforeUnloadEventHandlerNonNull? OnBeforeUnloadEventHandler;
 
 // https://html.spec.whatwg.org/multipage/#globaleventhandlers
-[NoInterfaceObject, Exposed=(Window,Worker)]
+[NoInterfaceObject, Exposed=Window]
 interface GlobalEventHandlers {
            attribute EventHandler onabort;
            attribute EventHandler onblur;
@@ -95,7 +95,7 @@ partial interface GlobalEventHandlers {
 };
 
 // https://html.spec.whatwg.org/multipage/#windoweventhandlers
-[NoInterfaceObject, Exposed=(Window,Worker)]
+[NoInterfaceObject, Exposed=Window]
 interface WindowEventHandlers {
            attribute EventHandler onafterprint;
            attribute EventHandler onbeforeprint;
@@ -115,7 +115,7 @@ interface WindowEventHandlers {
 };
 
 // https://html.spec.whatwg.org/multipage/#documentandelementeventhandlers
-[NoInterfaceObject, Exposed=(Window,Worker)]
+[NoInterfaceObject, Exposed=Window]
 interface DocumentAndElementEventHandlers {
           attribute EventHandler oncopy;
           attribute EventHandler oncut;

--- a/components/script/dom/webidls/Screen.webidl
+++ b/components/script/dom/webidls/Screen.webidl
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // http://dev.w3.org/csswg/cssom-view/#the-screen-interface
-[Exposed=(Window,Worker)]
+[Exposed=Window]
 interface Screen {
   //readonly attribute double availWidth;
   //readonly attribute double availHeight;

--- a/tests/wpt/mozilla/tests/mozilla/interfaces.worker.js
+++ b/tests/wpt/mozilla/tests/mozilla/interfaces.worker.js
@@ -9,7 +9,6 @@ importScripts("interfaces.js");
 test_interfaces([
   "Blob",
   "CloseEvent",
-  "CSSStyleDeclaration",
   "DOMMatrix",
   "DOMMatrixReadOnly",
   "DOMPoint",
@@ -38,7 +37,6 @@ test_interfaces([
   "ProgressEvent",
   "Request",
   "Response",
-  "Screen",
   "Storage",
   "StorageEvent",
   "TextDecoder",

